File: app/src/main/java/com/milenialsatwork/seriescalendar/model/data/NotaDAO.java
package com.milenialsatwork.seriescalendar.model.data;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class NotaDAO {

    private final static ArrayList<Series> notes = new ArrayList<>();

    public List<Series> todos() {
        return (List<Series>) notes.clone();
    }

    public void insert(Series... notes) {
        NotaDAO.notes.addAll(Arrays.asList(notes));
    }

    public void altera(int position, Series card) {
        notes.set(position, card);
    }

    public void remove(int position) {
        notes.remove(position);
    }

    public void swap(int starterPosition, int finalPosition) {
        Collections.swap(notes, starterPosition, finalPosition);
    }

    public void removeTodos() {
        notes.clear();
    }
}

File: app/src/main/java/com/milenialsatwork/seriescalendar/model/data/NotaDAO.kt
package com.milenialsatwork.seriescalendar.model.data

import com.milenialsatwork.seriescalendar.model.data.Series
import com.milenialsatwork.seriescalendar.model.data.NotaDAO
import java.util.*

class NotaDAO {
    fun todos(): List<Series> {
        return notes.clone() as List<Series>
    }

    fun insert(vararg notes: Series?) {
        Companion.notes.addAll(listOf(*notes))
    }

    fun altera(position: Int, card: Series?) {
        notes[position] = card
    }

    fun remove(position: Int) {
        notes.removeAt(position)
    }

    fun swap(starterPosition: Int, finalPosition: Int) {
        Collections.swap(notes, starterPosition, finalPosition)
    }

    fun removeTodos() {
        notes.clear()
    }

    companion object {
        private val notes = ArrayList<Series?>()
    }
}